src/ReactHardwareComponent.js

```diff
@@ -42,16 +64,23 @@ type ViewConfigPropTypes = {
  * @constructor ReactHardwareComponent
  * @extends ReactComponent
  * @extends ReactMultiChild
- * @param {!object} viewConfig View Configuration.
  */
-const ReactHardwareComponent = function(
-  element: React$Element
-) {
-  this.viewConfig = DEFAULT_VIEW_CONFIG;
+const ReactHardwareComponent = function(element: React$Element) {
   this._rootNodeID = null;
   this._renderedChildren = null;
   this._io = null;
-  this.construct(element);
+  this._currentElement = element;
+  this.viewConfig = DEFAULT_VIEW_CONFIG;
+  if (typeof element.type === 'string') {
+    this.viewConfig = getViewConfig(element.type);
+  }
+  if (process.env.NODE_ENV !== 'production') {
+    invariant(
```

Comment:
that may be more okay on RN where the thrown error happens on the JS thread and won’t (maybe won’t?) change the hardware output. If you’re rendering into hardware and an error gets thrown in the middle of some updates we may leave the hardware in an invalid/undesirable state. This is me “trying to do the right thing” in that scenario.